pkg/provisioner/ironic/ironic.go

```diff
@@ -625,102 +641,51 @@ func (p *ironicProvisioner) getUpdateOptsForNode(ironicNode *nodes.Node) (update
 				p.host.HardwareProfile()))
 	}
 
-	// image_source
-	var op nodes.UpdateOp
-	if _, ok := ironicNode.InstanceInfo["image_source"]; !ok {
-		op = nodes.AddOp
-		p.log.Info("adding image_source")
-	} else {
-		op = nodes.ReplaceOp
-		p.log.Info("updating image_source")
-	}
+	// instance_uuid
```

Comment:
I'd have assumed we only want to set this when provisioning ('instance' here means Nova instance).

Reply:
IIUC, ironic doesn't use any of these values until it needs them, so setting them early shouldn't hurt. They're reset before provisioning, as well.